### PR TITLE
Fix `awaitData` cursors

### LINKED ISSUE #3945: Fix flaky tests for tailable cursors
### What should be done?

https://github.com/FerretDB/FerretDB/actions/runs/7431286062/job/20221864094?pr=3941

```
FAIL TestCursorsTailableAwaitData/GetMoreNewDoc (0.07s/0.07s):
=== RUN   TestCursorsTailableAwaitData/GetMoreNewDoc
    awaitdata_test.go:291: 
        	Error Trace:	/home/runner/work/FerretDB/FerretDB/integration/cursors/awaitdata_test.go:291
        	Error:      	Not equal: 
        	            	expected: &types.Document{keys:map[string]int{"_id":1}, fields:[]types.field{types.field{value:"new", key:"_id"}}, frozen:false, recordID:0}
        	            	actual  : &types.Document{keys:map[string]int{"_id":1}, fields:[]types.field{types.field{value:2, key:"_id"}}, frozen:false, recordID:0}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -6,3 +6,3 @@
        	            	   (types.field) {
        	            	-   value: (string) (len=3) "new",
        	            	+   value: (int32) 2,
        	            	    key: (string) (len=3) "_id"
        	Test:       	TestCursorsTailableAwaitData/GetMoreNewDoc
--- FAIL: TestCursorsTailableAwaitData/GetMoreNewDoc (0.07s)

FAIL TestCursorsTailableAwaitData/GetMoreEmptyAfterInsertion (0.03s/0.03s):
=== RUN   TestCursorsTailableAwaitData/GetMoreEmptyAfterInsertion
    awaitdata_test.go:306: 
        	Error Trace:	/home/runner/work/FerretDB/FerretDB/integration/cursors/awaitdata_test.go:306
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestCursorsTailableAwaitData/GetMoreEmptyAfterInsertion
--- FAIL: TestCursorsTailableAwaitData/GetMoreEmptyAfterInsertion (0.03s)
```

### Where?

https://github.com/FerretDB/FerretDB/blob/main/integration/cursors/awaitdata_test.go

### Definition of Done

Tests are not flaky anymore.

### LINKED ISSUE #3957: Tailable cursor with `awaitData` should block and not error
### FerretDB version

v1.18.0-2-g2c2db29b

### Backend

only checked with PostgreSQL

### Environment

dev

### What did you do?

See #3956.

### What did you expect to see?

After the timeout expired an empty batch and an open cursor.

```
        "cursor" : {
                "nextBatch" : [ ],
                "id" : NumberLong("7129611981940357923"),
                "ns" : "test.foo"
        }
```

### What did you see instead?

```
uncaught exception: Error: command failed: {
        "ok" : 0,
        "errmsg" : "[msg_getmore.go:253 handler.(*Handler).MsgGetMore] [msg_getmore.go:332 handler.(*Handler).awaitData.func1] [cursor.go:121 cursor.(*Cursor).Reset] [projection_iterator.go:60 common.(*projectionIterator).Next] [filter_iterator.go:52 common.(*filterIterator).Next] [query_iterator.go:95 postgresql.(*queryIterator).Next] iterator is read to the end or closed",
        "code" : 1,
        "codeName" : "InternalError"
} with original command request: {
        "getMore" : NumberLong(62),
        "collection" : "foo",
        "batchSize" : 1,
        "maxTimeMS" : 60000,
        "lsid" : {
                "id" : UUID("3574cdf1-b409-40e7-b61c-596cb64eb407")
        }
```

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #3957.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by AlekSi — COMMENTED** [review]
CI fails after merging #3940

File: integration/cursors/awaitdata_test.go
Comment (AlekSi) [inline integration/cursors/awaitdata_test.go:330]: We are not ready with the setup yet at this point. This should be moved lower

File: integration/cursors/awaitdata_test.go
Comment (AlekSi) [inline integration/cursors/awaitdata_test.go:313]: We should link to issues, not to PRs.

And can we use expected failure instead?

File: integration/cursors/awaitdata_test.go
Comment (AlekSi) [inline integration/cursors/awaitdata_test.go:321]: `t.Run` doesn't run in parallel, effectively un-stressing the test

File: integration/cursors/awaitdata_test.go
Comment (AlekSi) [inline integration/cursors/awaitdata_test.go:358]:
```suggestion
			var firstBatch *types.Array
			firstBatch, cursorID := getFirstBatch(t, res)
```

There is no need to define `cursorID` above either.

(I wonder why everyone suddenly started using `var` instead of short variables assignments…)

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: integration/cursors/awaitdata_test.go
Comment (noisersup) [reply integration/cursors/awaitdata_test.go]: From my testing it seems it's not

File: integration/cursors/awaitdata_test.go
Comment (noisersup) [reply integration/cursors/awaitdata_test.go]: Most of the subtests pass, and usually one of them fails, so not really

File: integration/cursors/awaitdata_test.go
Comment (AlekSi) [reply integration/cursors/awaitdata_test.go:321]: Ok, please explain how `t.Run` works then

**Comment by AlekSi** [comment]
On hold for now

**Pull request closed without merging**
